server/src/main/java/org/opensearch/snapshots/RestoreService.java

```diff
@@ -763,12 +836,17 @@ private IndexMetadata updateIndexSettings(
                         Settings.Builder settingsBuilder = Settings.builder()
                             .put(settings.filter(settingsFilter))
                             .put(normalizedChangeSettings.filter(k -> {
-                                if (UNMODIFIABLE_SETTINGS.contains(k)) {
+                                if (USER_UNMODIFIABLE_SETTINGS.contains(k)) {
                                     throw new SnapshotRestoreException(snapshot, "cannot modify setting [" + k + "] on restore");
                                 } else {
                                     return true;
                                 }
                             }));
+
+                        // override internal settings
+                        if (changeSettingsInternal != null) {
```

Comment:
do we even need to this? if we just ignore remote store and replication type settings, then wouldn't `MetadataCreateIndexService ` put in the correct settings automatically?